### PR TITLE
Deployment of new ONT workflow

### PR DESCRIPTION
**Pull request by kedhammar**

### CODE REVIEW

File: scripts/ont_calc_volumes.py [hunk 1]
```diff
@@ -3,88 +3,176 @@
...
+                for f in out.files:
+                    lims.request_session.delete(f.uri)
+                lims.upload_new_file(out, log_filename)
 
-            if "Amount (ng)" in [t[0] for t in art_tuple[1]["uri"].udf.items()]:
-                art_tuple[1]["uri"].udf["Amount (ng)"] = round(amt_taken_ng,1)
-            art_tuple[1]["uri"].udf["Amount (fmol)"] = round(amt_taken_fmol,1)
-            art_tuple[1]["uri"].udf["Volume to take (uL)"] = round(vol_to_take,1)
-            art_tuple[1]["uri"].put()
+    except BaseException as e:
```
Comment (alneberg) [inline scripts/ont_calc_volumes.py:166]: I would probably move this try except outside the `main` 

```python
try:
    main(lims, args)
except BaseException as e:
    sys.stderr.write(str(e))
    sys.exit(2)

File: scripts/ont_calc_volumes.py [hunk 2]
```diff
@@ -3,88 +3,176 @@
...
+            art_out = art_tuple[1]["uri"]
+            log.append(f"Input {art_in.name} --> Output {art_out.name}")
 
-            # Calculate
-            if basis == "fmol" or basis == "ng":
+            # Get last known length
+            size_bp, size_bp_history = udf_tools.fetch_last(
+                currentStep, art_tuple, "Size (bp)", on_fail=None, print_history=True
+            )
+            log.append(f"'Size (bp): {size_bp}\n{size_bp_history}")
```
Comment (alneberg) [inline scripts/ont_calc_volumes.py:48]: Unclosed string `'`?

Comment (kedhammar) [reply scripts/ont_calc_volumes.py]: [a7fbe27](https://github.com/SciLifeLab/scilifelab_epps/pull/250/commits/a7fbe278ee1f705687931e63c9d5d7f8717e9f0c)

File: scripts/ont_calc_volumes.py (hunk 1 above)
Comment (kedhammar) [reply scripts/ont_calc_volumes.py]: [a7fbe27](https://github.com/SciLifeLab/scilifelab_epps/pull/250/commits/a7fbe278ee1f705687931e63c9d5d7f8717e9f0c)

File: scripts/ont_send_loading_info_to_db.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+                "load_fmol": art.udf["ONT flow cell loading amount (fmol)"],
+            }
+
+            try:
+                try:
+                    doc["lims"]["loading"].append(dict_to_add)
+                except KeyError:
+                    doc["lims"]["loading"] = []
+                    doc["lims"]["loading"].append(dict_to_add)
+            except:
```
Comment (alneberg) [inline scripts/ont_send_loading_info_to_db.py:82]: I think a 
```python
if "lims" not in doc:
    doc["lims"] = {}
``` 
Before this section would be slightly nicer?

File: scripts/ont_send_loading_info_to_db.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+            dict_to_add = {
+                "step_name": currentStep.type.name,
+                "pid": currentStep.id,
+                "timestamp": timestamp,
+                "qc": row.qc_pore_count,
+                "load_fmol": row.initial_loading_fmol,
+            }
+
+            try:
```
Comment (alneberg) [inline scripts/ont_send_loading_info_to_db.py:225]: Same comment here as above

File: scripts/ont_send_reloading_info_to_db.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+def main(lims, args):
+    """For all samples/flowcells, use the run name to find the correct database entry.
+
+    Then update the document "lims" json object nest with the reloading information.
+
+    TODO Get parent process ID on a sample-by-sample basis, rather than once for the entire step.
+         Current approach may cause issues if samples originate from different steps.
+    """
+
+    try:
```
Comment (alneberg) [inline scripts/ont_send_reloading_info_to_db.py:31]: Same comment as before

File: scripts/ont_send_reloading_info_to_db.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+                dict_to_add = {
+                    "step_name": currentStep.type.name,
+                    "pid": currentStep.id,
+                    "timestamp": timestamp,
+                    "reload_times": run["reload_times"],
+                    "reload_fmols": run["reload_fmols"],
+                    "reload_lots": run["reload_lots"],
+                }
+
+                try:
```
Comment (alneberg) [inline scripts/ont_send_reloading_info_to_db.py:77]: Here as well

**Review by alneberg — APPROVED** [review]
Amazing work! Both you and black, 👍

File: scripts/ont_send_loading_info_to_db.py (hunk 3 above)
Comment (kedhammar) [reply scripts/ont_send_loading_info_to_db.py]: [2131d7b](https://github.com/SciLifeLab/scilifelab_epps/pull/250/commits/2131d7bf409897642ec1ad486830494dd3a3f153)

File: scripts/ont_send_reloading_info_to_db.py (hunk 6 above)
Comment (kedhammar) [reply scripts/ont_send_reloading_info_to_db.py]: [cf18f5b](https://github.com/SciLifeLab/scilifelab_epps/pull/250/commits/cf18f5bc1e5df13b3fc35a159bd9991b64b47651)

File: scripts/ont_send_reloading_info_to_db.py (hunk 5 above)
Comment (kedhammar) [reply scripts/ont_send_reloading_info_to_db.py]: [cf18f5b](https://github.com/SciLifeLab/scilifelab_epps/pull/250/commits/cf18f5bc1e5df13b3fc35a159bd9991b64b47651)

File: scripts/ont_send_loading_info_to_db.py (hunk 4 above)
Comment (kedhammar) [reply scripts/ont_send_loading_info_to_db.py]: [cf18f5b](https://github.com/SciLifeLab/scilifelab_epps/pull/250/commits/cf18f5bc1e5df13b3fc35a159bd9991b64b47651)

**Pull request merged**